admin-auth/src/main/kotlin/com/epam/drill/admin/auth/service/impl/PasswordServiceImpl.kt

```diff
@@ -28,7 +28,8 @@ class PasswordServiceImpl(
         return BCrypt.hashpw(password, BCrypt.gensalt())
     }
 
-    override fun matchPasswords(candidate: String, hashed: String): Boolean {
+    override fun matchPasswords(candidate: String, hashed: String?): Boolean {
+        if (hashed == null) return false
```

Comment:
push `if (hashed == null)` check out of this function. `matchPasswords` should not be executed for oauth user

Reply:
think about more

Reply:
- I think the password matching service works with null hashes as expected and does not take on unnecessary responsibility. 
- Сhecking for null outside the matching service will be a little more difficult than just checking for null, because the passwordHash field has a var (not a val) definition
- I don’t think that in the simple authentication service we need to explicitly check for the provider through which the user was registered, we just need to make sure that we won’t be able to log in with any password

Reply:
> In any case, I simplified the implementation a little and maybe in this form it raises fewer questions:
> 
> ```
>     override fun matchPasswords(candidate: String, hashed: String?): Boolean {
>         return hashed != null && BCrypt.checkpw(candidate, hashed)
>     }
> ```

I would it raises pretty much the same questions 😄 But I agree with your points above. Resolving that conversation